src/mpfao/2D/tdympfao2D.c

```diff
@@ -69,7 +70,7 @@ PetscErrorCode TDyComputeGMatrixFor2DMesh(TDy tdy) {
     // extract permeability tensor
     for (ii=0; ii<dim; ii++) {
       for (jj=0; jj<dim; jj++) {
-        K[ii][jj] = tdy->K[icell*dim*dim + ii*dim + jj];
+        K[ii][jj] = matprop->K[icell*dim*dim + ii*dim + jj];
```

Comment:
If we intend the fields of a MaterialProp structure to be directly manipulated, I don't think we should make its underlying type a pointer. The pointer pattern makes sense when you manipulate the data of an "object" only through functions that belong to its interface.

The reason is this: here we see that a variable is declared as what seems to be a scalar type, and then it's treated as a pointer (with `->` to access its fields). This looks awfully strange to a C programmer. If it's really just a collection of fields to be accessed freely without an interface, I recommend that we use what we're calling `_MaterialProp` as `MaterialProp`.

Reply:
If we redefine `MaterialProp` as just a struct, line 64 changes to

```
MaterialProp* matprop = tdy->matprop;
```
This makes it clear that we're using a pointer, and that the fields should be accessed via `->`.